eskipfile/remote_test.go

```diff
@@ -91,6 +91,13 @@ func TestLoadAll(t *testing.T) {
 		t.Run(test.title, func(t *testing.T) {
 			options := &RemoteWatchOptions{RemoteFile: s.URL, Threshold: 10, Verbose: true, FailOnStartup: true}
 			client, err := RemoteWatch(options)
+			defer func() {
+				c, ok := client.(*remoteEskipFile)
+				if ok {
```

Comment:
Do we need ok check here? I think its better for test to fail if the type ever changes

Reply:
I see now `RemoteWatch` API does not allow to close returned client :(

Reply:
I still think `client.(*remoteEskipFile).Close()` would be better

Reply:
why is this important?

Reply:
To let if fail if client type changes in the future

Reply:
I get:
```
% go test ./eskipfile        
--- FAIL: TestLoadAll (0.00s)
    --- FAIL: TestLoadAll/Download_not_existing_remote_file_fails_in_NewRemoteEskipFile (0.00s)
panic: interface conversion: routing.DataClient is nil, not *eskipfile.remoteEskipFile [recovered]
        panic: interface conversion: routing.DataClient is nil, not *eskipfile.remoteEskipFile
..
github.com/zalando/skipper/eskipfile.TestLoadAll.func1(0xc000103520)
        /home/sszuecs/go/src/github.com/zalando/skipper/eskipfile/remote_test.go:94 +0x3d1 fp=0xc000409f70 sp=0xc000409ea0 pc=0xe0c151
testing.tRunner(0xc000103520, 0xc000118960)
```

Reply:
I have a fix: check `err == nil` before running the defer Close :)

Reply:
Need to move defer after err checking

Reply:
What? See `==` vs `!=`, I know not the Go obvious to check `==`, but it's the simplest to do.

Reply:
Ok :)